ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6324,6 +6324,20 @@ void stream_xprt_term(ldms_t x)
 		__RSE_free(ent);
 	}
 	__RSE_rbt_unlock();
+
+	/* Free outstanding configuration requests */
+	req_ctxt_tree_lock();
+	ldmsd_req_ctxt_t reqc;
+	RBT_FOREACH(rbn, &msg_tree) {
+		struct rbn *succ;
+		reqc = container_of(rbn, struct ldmsd_req_ctxt, rbn);
+		if (reqc->xprt->ldms.ldms == x) {
```

Comment:
@nichamon did you say that this check is incorrect or could collide with multiple transports?

Reply:
No, I didn't say that. What I mentioned was the `rbn` at line 6337 is not evaluated because the macro `RBT_FOREACH` will make `rbn` pointing to its successor in the next iteration. We may replace the macro with 
```
rbn = rbt_min(&msg_tree);
while (rbn) {
     succ = rbn_succ(rbn);
     /* Cleanup the request context if it corresponds to x. */     
     rbn = succ;
}
```

Reply:
Another thing is that only the passive side of the connection calls `ldmsd_xprt_term()`. The active side could leak request contexts in `msg_tree` that it has not received the EOF of the request yet. This will occur when a sampler daemon republished a large stream to an aggregator, but the aggregator has not received all the records.

Reply:
Ah right! Thanks, I remembered that you had issues with this but I couldn't remember what they were.